user-org-es-util/src/main/java/org/sunbird/es/service/impl/ElasticSearchRestHighImpl.java

```diff
@@ -0,0 +1,652 @@
+    return promise.future();
+  }
+
+  /**
+   * This method will remove data from ES based on identifier.
+   *
+   * @param index String
+   * @param type String
+   * @param identifier String
+   */
+  @Override
+  public Future<Boolean> delete(String index, String identifier) {
+    long startTime = System.currentTimeMillis();
+    ElasticSearchHelper.logMethodStart(
+        "ElasticSearchRestHighImpl:delete: method started at ==", startTime);
+    Promise<Boolean> promise = Futures.promise();
+    DeleteRequest delRequest = new DeleteRequest(index, _DOC, identifier);
+    ActionListener<DeleteResponse> listener =
+        new ActionListener<DeleteResponse>() {
+          @Override
+          public void onResponse(DeleteResponse deleteResponse) {
+            if (deleteResponse.getResult() == DocWriteResponse.Result.NOT_FOUND) {
+              ProjectLogger.log(
+                  "ElasticSearchRestHighImpl:delete:OnResponse: Document  not found for index : "
+                      + index
+                      + " , identifier : "
+                      + identifier,
+                  LoggerEnum.INFO.name());
+
+              promise.success(false);
+            } else {
+              promise.success(true);
+            }
+          }
+
+          @Override
+          public void onFailure(Exception e) {
+
+            ProjectLogger.log(
+                "ElasticSearchRestHighImpl:delete: Async Failed due to error :" + e,
+                LoggerEnum.INFO.name());
+
+            promise.failure(e);
+          }
+        };
+
+    ConnectionManager.getRestClient().deleteAsync(delRequest, listener);
+    ElasticSearchHelper.logMethodEnd("ElasticSearchRestHighImpl:delete: method end ==", startTime);
+    return promise.future();
+  }
+
+  /**
+   * Method to perform the elastic search on the basis of SearchDTO . SearchDTO contains the search
+   * criteria like fields, facets, sort by , filters etc. here user can pass single type to search
+   * or multiple type or null
+   *
+   * @param type var arg of String
+   * @return search result as Map.
+   */
+  @Override
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public Future<Map<String, Object>> search(SearchDTO searchDTO, String index) {
+    long startTime = System.currentTimeMillis();
+    ElasticSearchHelper.logMethodStart(
+        "ElasticSearchRestHighImpl:search: method started at ==", startTime);
+    SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+    SearchRequest searchRequest = new SearchRequest(index);
+    searchRequest.types(_DOC);
+
+    // check mode and set constraints
+    Map<String, Float> constraintsMap = ElasticSearchHelper.getConstraints(searchDTO);
+
+    BoolQueryBuilder query = new BoolQueryBuilder();
+
+    // add channel field as mandatory
+    String channel = PropertiesCache.getInstance().readProperty(EsConstant.SUNBIRD_ES_CHANNEL);
+    if (!(StringUtils.isBlank(channel) || EsConstant.SUNBIRD_ES_CHANNEL.equals(channel))) {
```

Comment:
If we are using SUNBIRD_ES_CHANNEL, then all env variable should be at one place(externalResource properties file)

Reply:
This is removed now.